### PR TITLE
Abort a build if a newer one is scheduled 🛑

### PR DESCRIPTION
**Pull request by ivopintodasilva**
If a new build is found on queue for the current job, abort the current build.

### CODE REVIEW

**Review by p4checo — APPROVED** [review]
Great addition! 👏😄

**Comment by p4checo** [comment]
My only suggestion would be to try and make the name shorter, e.g. `abortBuildIfNewerExists`?

**Comment by beloso** [comment]
What's the difference from the milestone step?
Also, will this action release any resource locks?

**Review by p4checo — CHANGES_REQUESTED** [review]
Just remembered that you're missing an entry on README.md 🚓 👮 😇

**Comment by ivopintodasilva** [comment]
@beloso I didn't know about the milestone step plugin but it has a couple of differences, from what I've understood:

- Older builds will not proceed (they are aborted) if a newer one already passed the milestone;
- When a build passes a milestone, any older build that passed the previous milestone but not this one is aborted.

If we have several builds executing concurrently and want to keep only the newest build (killing all the old ones), the milestone plugin is great.

However, if we have a build running and our executors are full, when we schedule a new build (even if it's newer than the one that is already running), we have to wait until the executors are available to run our build in order to abort older builds based on milestones.

Our approach is that, at each stage, the running build checks if it is the newest build available. If it is not, it aborts and frees the executor for the builds that are on the queue.
We are trying to free the executors as soon as possible and not so much solve a concurrency problem 🙂

**Comment by ivopintodasilva** [comment]
Renamed and added action to README @p4checo 🙂

File: README.md [hunk 1]
```diff
@@ -102,3 +102,23 @@ stage('Demo') {
...
 ```
+
+### abortBuildIfNewerExists
+
+Action that aborts a build if the current job has a newer build to execute.
+
+Parameter | Type | Default value | Description
+------------ | ------------- | ------------- | -------------
+description | `String` | `"action"` | description of the action (for logging purposes)
+
```
Comment (p4checo) [inline README.md:135]: please add a "Return" section too 😛

Comment (ivopintodasilva) [reply README.md:135]: Added 😄

File: README.md [hunk 2]
```diff
@@ -120,3 +126,27 @@ stage('Demo') {
...
+N/A
+
+#### Example
+
+```groovy
+@Library('jenkins-pipeline-library')_
+
+stage('Demo') {
+
+  abortBuildIfNewerExists(description: "'abort build if newer exists'")
```
Comment (p4checo) [inline README.md]: you forgot to remove the description from the example 😛

**Review by p4checo — APPROVED** [review]
🚀

**Pull request merged**